test/to-json.test.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const test = require('tap').test
+const boot = require('..')
+
+test('to json', (t) => {
+  t.plan(4)
+
+  const app = boot()
+  app
+    .use(one)
+    .use(two)
+    .use(three)
+
+  const outJson = {
+    avvio: {
+      label: 'avvio',
+      start: /\d*/,
+      stop: /\d*/,
+      diff: /\d*/,
+      children: []
+    }
+  }
+
+  app.on('preReady', function show () {
+    const json = app.toJSON()
+    t.like(json, outJson)
+  })
+
+  function one (s, opts, done) {
+    const json = app.toJSON()
+    outJson.avvio.children.push({
+      label: 'one',
+      start: /\d*/,
+      stop: /\d*/,
+      diff: /\d*/
+    })
+    t.like(json, outJson)
+    done()
+  }
+  function two (s, opts, done) {
+    const json = app.toJSON()
+    outJson.avvio.children.push({
+      label: 'two',
+      start: /\d*/,
+      stop: /\d*/,
+      diff: /\d*/
+    })
+    t.like(json, outJson)
+    done()
+  }
+  function three (s, opts, done) {
+    const json = app.toJSON()
+    outJson.avvio.children.push({
+      label: 'three',
+      start: /\d*/,
+      stop: /\d*/,
+      diff: /\d*/
+    })
+    t.like(json, outJson)
+    done()
+  }
+})
```

Comment:
Can you add another test with a multi-level hierarchy of plugins?

Reply:
I'm not sure: what do you expect?
Because I get the same output as a single level: in fact, the plugins are all loaded sequentially

Reply:
I think the goal should be to show a tree, not just the loading order.

Reply:
In other terms, the information that two was loaded from one is important.

Reply:
fixed
```
bound root 3 ms
└─┬ first 3 ms
  ├── second 0 ms
  └─┬ third 2 ms
    └── fourth 1 ms
```

But I have to check way in fastify/example I get:

```$ node examples/example.js
bound root 11 ms
├── bound _after 5 ms
├── bound _after undefined ms
├── bound _after undefined ms
├── bound _after undefined ms
├── bound _after undefined ms
├── bound _after undefined ms
├── bound _after undefined ms
├── bound _after undefined ms
└── bound _after undefined ms
```

Reply:
That is more or less what it should be happening :).
Here is the culprit: https://github.com/fastify/fastify/blob/356cf6f078a9f93b009ddb6384deaffe549d1032/fastify.js#L589.

I don't know how we would like to represent those.